src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -5176,7 +5176,9 @@ public static Map groupBy(Object[] self, List<Closure> closures) {
             G key = outer.getKey();
             List<Map.Entry<K, V>> entries = outer.getValue();
             Map<K, V> target = createSimilarMap(self);
-            putAll(target, entries);
```

Comment:
Yes, but feel free to play around with other options if you have a strong preference.